src/hotspot/share/adlc/output_c.cpp

```diff
@@ -3985,7 +3985,8 @@ void ArchDesc::buildMachNode(FILE *fp_cpp, InstructForm *inst, const char *inden
 
   // Fill in the bottom_type where requested
   if (inst->captures_bottom_type(_globalNames)) {
-    if (strncmp("MachCall", inst->mach_base_class(_globalNames), strlen("MachCall"))) {
+    if (strncmp("MachCall", inst->mach_base_class(_globalNames), strlen("MachCall"))
+      && strncmp("MachIf", inst->mach_base_class(_globalNames), strlen("MachIf"))) {
```

Comment:
Maybe explicitly compare the results with 0 here for consistency with other usage of `strcmp`

Reply:
Rethink about it should it be `strncmp("MachCall", inst->mach_base_class(_globalNames), strlen("MachCall") + 1` instead? Or else we could return true for nodes that have `MachCall` as a prefix.

Reply:
@haominw Ah I understand, is this behaviour similarly expected for `MachIf`, or we want the node to exactly match here? Thanks.

Reply:
Just want to match MachIfNode. Because the output of `inst->mach_base_class(_globalNames)` is only `MachIfNode`, no other `MachIf`, so I just match `MachIf`.

Reply:
In that case it should be `strlen("MachIf") + 1` since you need to compare the null terminated character, too.